src/Featurizers/Components/RollingWindowTransformerBase.h

```diff
@@ -0,0 +1,163 @@
+///                 to be implemented by the derived classes.
+///
+template <
+    typename InputT,
+    typename OutputT,
+    size_t MaxNumTrainingItemsV
+>
+class RollingWindowTransformerBase:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>>;
+    using CalculatorFunctionInputType = typename Components::CircularBuffer<InputT>::iterator;
+    using CalculatorFunction = std::function<typename OutputT(typename CalculatorFunctionInputType, typename CalculatorFunctionInputType)>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    RollingWindowTransformerBase(std::uint32_t windowSize, CalculatorFunction calculator, std::uint32_t horizon, std::uint32_t minWindowSize) ;
+    ~RollingWindowTransformerBase(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(RollingWindowTransformerBase);
+
+    bool operator==(RollingWindowTransformerBase const &other) const;
+    bool operator!=(RollingWindowTransformerBase const &other) const;
+
+protected:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Protected Members
+    // |
+    // ----------------------------------------------------------------------
+    // These are protected so that the derived classes can use them for saving of state.
+    const std::uint32_t                             _windowSize;
+    const std::uint32_t                             _horizon;
+    const std::uint32_t                             _minWindowSize;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Members
+    // |
+    // ----------------------------------------------------------------------
+    Components::CircularBuffer<InputT>              _buffer;
+    CalculatorFunction                              _calculator;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        typename BaseType::TransformedType         results(_horizon);
+        
+        _buffer.push(input);
+        size_t bufferSize = _buffer.size();
+
+        OutputT result;
+        size_t numElements;
+        size_t startingOffset;

```

Comment:
These aren't used outside of the loop so they should be defined local within the loop.

Reply:
Moved into the loop.